runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/FieldTemplate/AdditionalPropertiesField.tsx

```diff
@@ -0,0 +1,110 @@
+    const additional = schema.hasOwnProperty(ADDITIONAL_PROPERTY_FLAG);
+    const { translate } = useTranslations();
+    const [mainTitle, setMainTitle] = React.useState(schema?.mainTitle);
+    const [subTitle, setSubTitle] = React.useState(schema?.subTitle);
+    const [mainTitleTranslationKey, setMainTitleTranslationKey] = React.useState<string|boolean>(false);
+    const [subTitleTranslationKey, setSubTitleTranslationKey] = React.useState<string|boolean>(false);
+
+    useEffect(() => {
+        if (!mainTitleTranslationKey && !subTitleTranslationKey && mainTitle && subTitle) {
+            setMainTitleTranslationKey(`Process.Details.Modeler.DetailsSidebar.${capitalizeFirstLetter({ text: mainTitle })}.Label`);
+            setSubTitleTranslationKey(`Process.Details.Modeler.DetailsSidebar.${capitalizeFirstLetter({ text: subTitle })}.Label`);
+        }
+        if (mainTitleTranslationKey && subTitleTranslationKey) {
+            // @ts-ignore
+            setMainTitle(translate(mainTitleTranslationKey));
+            // @ts-ignore
+            setSubTitle(translate(subTitleTranslationKey));
+        }
+    }, [mainTitle, subTitle, mainTitleTranslationKey, subTitleTranslationKey, i18n.language]);
+
+    if (!additional) 
+    { return <>{children}</>; }
+
+    const handleBlur = ({ target }: React.FocusEvent<HTMLInputElement>) => onKeyChange(target.value);
+
+    const formData = children.props.children[0].props.children[0].props.formData;
+
+    return (
+        <Grid container key={`${id}-key`} alignItems="center" spacing={1}>
+            <Grid item xs={12}>
+                <TextField
+                    fullWidth
+                    required={required}
+                    variant="outlined"
+                    label={mainTitle ? mainTitle : 'Key'}
+                    size="medium"
+                    defaultValue={label !== 'newKey' && label !== 'additionalProperties' ? label : ''}
+                    disabled={disabled || readonly}
+                    id={`${id}-key`}
+                    name={`${id}-key`}
+                    onBlur={!readonly ? handleBlur : undefined}
+                    type="text"
+                />
+            </Grid>
+            <Grid item xs={12}>
+                <ElementAwareAutocompleteWidget
+                    {...children.props.children[0].props.children[0].props}
+                    required={required}
+                    label={subTitle ? subTitle : 'Value'}
+                    defaultValue={formData !== 'New Value' && typeof formData === 'string' ? formData : ''}
+                    value={formData !== 'New Value' && typeof formData === 'string' ? formData : ''}
+                    disabled={disabled || readonly}
+                    type="text"
+                />
+            </Grid>
+            <Grid item>
+                <Button color="secondary" disabled={disabled || readonly} onClick={onDropPropertyClick(label)}>
+                    <Remove />
+                    {' '}
```

Comment:
use flex with gap instead of just a space, this is the way

Reply:
{' '} was added by mistake and it has no influence on space between icon and text of the button.

But I'll bury your advice deep in my mind.